Core/source/core/pgp-key.ts

```diff
@@ -135,6 +135,13 @@ export class PgpKey {
     return p instanceof SecretKeyPacket || p instanceof SecretSubkeyPacket;
   }
 
+  public static validateAllDecryptedPackets = async (key: Key): Promise<void> => {
+    const packets = key.toPacketList() as PacketList<SecretKeyPacket>;
```

Comment:
Please remove type cast to `PacketList<SecretKeyPacket>`, not all packets will be of type `SecretKeyPacket`, return type of the `toPacketList()` is `PacketList<AllowedKeyPackets>`, that's what we need to operate on.

Reply:
thank you, sir, I revert the change to use `PacketList<AllowedKeyPackets>` and check if the instance of `prvPacket` is SecretKeyPacket and if the prvPacket is already decrypted because I don't have much idea of how I could make the `.filter(PgpKey.isPacketPrivate)` to work.

Reply:
What's the problem with `.filter(PgpKey.isPacketPrivate)`? Why doesn't it work?

Reply:
I'm not sure but i'm getting the following type error when doing so:
```
No overload matches this call.
  Overload 1 of 2, '(predicate: (value: AllowedKeyPackets, index: number, array: AllowedKeyPackets[]) => value is PrvPacket, thisArg?: any): PrvPacket[]', gave the following error.
    Argument of type '(p: AnyKeyPacket) => p is PrvPacket' is not assignable to parameter of type '(value: AllowedKeyPackets, index: number, array: AllowedKeyPackets[]) => value is PrvPacket'.
      Types of parameters 'p' and 'value' are incompatible.
        Type 'AllowedKeyPackets' is not assignable to type 'BasePublicKeyPacket'.
          Type 'SignaturePacket' is missing the following properties from type 'BasePublicKeyPacket': algorithm, getAlgorithmInfo, getFingerprint, getFingerprintBytes, and 6 more.
  Overload 2 of 2, '(predicate: (value: AllowedKeyPackets, index: number, array: AllowedKeyPackets[]) => unknown, thisArg?: any): AllowedKeyPackets[]', gave the following error.
    Argument of type '(p: AnyKeyPacket) => p is PrvPacket' is not assignable to parameter of type '(value: AllowedKeyPackets, index: number, array: AllowedKeyPackets[]) => unknown'.
      Types of parameters 'p' and 'value' are incompatible.
        Type 'AllowedKeyPackets' is not assignable to type 'BasePublicKeyPacket'.ts(2769)
```

Reply:
Maybe just change parameter type of `isPrvPacket()` to `AllowedKeyPackets`?

Reply:
@martgil I've changed parameter type of the `isPacketPrivate()` to `AllowedKeyPackets` and it worked. I've pushed this change.